tests/installation/first_boot.pm

```diff
@@ -25,7 +25,10 @@ sub handle_login {
     if (get_var('DM_NEEDS_USERNAME')) {
         type_string $username;
     }
-    send_key "ret";
+    if (check_var('DESKTOP', 'gnome') || get_var('DM_NEEDS_USERNAME')) {
```

Comment:
I suggest to move the ret into the above if as 'type_string "$username\n"'; - and have the send_key only for gnome. This can also be replaced by check_screen 'gdm' somewhen if necessary (i.e. in an event of xfce switching to it)